packages/inferno-router/src/MemoryRouter.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * @module Inferno-Router
+ */ /** TypeDoc Comment */
+
+import { createVNode, VNode, Component } from "inferno";
+import VNodeFlags from "inferno-vnode-flags";
+import createHistory from "history/createMemoryHistory";
+import Router from "./Router";
+import { warning } from "./utils";
+
+export interface IMemoryRouterProps {
+  initialEntries: string[];
+  initialIndex: number;
+  getUserConfirmation: () => {};
+  keyLength: number;
+  children: Component<any, any>[];
+}
+
+export default class MemoryRouter extends Component<IMemoryRouterProps, any> {
+  public history;
+
+  constructor(props?: any, context?: any) {
+    super(props, context);
+    this.history = createHistory(props);
+  }
+
+  public componentWillMount() {
+    warning(
```

Comment:
this componentWillMount function could be behind process.env.NODE_ENV !== 'production'  to reduce production size of library